docs/02-guides/prevent-excessive-usage-with-rate-limits.mdx

```diff
@@ -0,0 +1,186 @@
+---
+title: Prevent excessive usage with rate limits
+description:
+  This guide explores the implementation of rate limiting techniques to
+  safeguard your applications from excessive use.
+---
+
+<p>{frontMatter.description}</p>
+
+## Rate Limiter Service
+
+The Rate Limiter service provides a robust solution for managing and preventing
+excessive usage by imposing limits on the number of requests that users can make
+within a specified period. This service utilizes Redis for storage and supports
+integration into both HTTP contexts and GraphQL operations.
+
+### Features
+
+- **Redis-backed Storage**: Utilizes Redis to efficiently track request counts
+  and support high concurrency.
+- **Flexible Rate Limits**: Configure max requests and duration per rate limit
+  namespace.
+- **HTTP and GraphQL Support**: Easily integrate rate limiting into Remix
+  actions/loaders and GraphQL resolvers.
+- **Automated Response Handling**: Automatically sends HTTP 429 Too Many
+  Requests responses when a limit is exceeded.
+
+### Service Configuration
+
+The rate limiter requires a Redis configuration to function. Upon instantiation,
+it subscribes to a Redis instance and sets up namespaces for different rate
+limits.
+
+To configure the service, you can add the following to your
+`front-commerce.config.ts`:
+
+```ts title="front-commerce.config.ts"
+import { defineConfig } from "@front-commerce/core/config";
+
+export default defineConfig({
+  // ...other configuration options
+  rateLimiter: {
+    redis: {
+      host: process.env.FRONT_COMMERCE_CLOUD_REDIS_SESSIONS_HOST,
+      port: process.env.FRONT_COMMERCE_CLOUD_REDIS_SESSIONS_PORT || 6379,
+      db: process.env.FRONT_COMMERCE_CLOUD_REDIS_SESSIONS_DB || 2,
+    },
+  },
+});
+```
+
```

Comment:
**issue:** I think it's important to highlight that we recommend to shate the same Redis connection than sessions, hence our suggestion, but that it's not required.

**suggestion:**  something like that (to be refined as you prefer)

```suggestion
:::tip

We've suggested a configuration which shares the same Redis DB than sessions. While it's not required, we think it makes sense because both share the same kind of responsibilities.

:::
```